File: webhooks/utils/agent_task.py
"""
Workflow adapter for invoking a deployed Conductor agent. This allows an AGENT task to be
included in a workflow by using the ">>" operator to chain it with non-AGENT tasks.
"""

from conductor.client.http.models import WorkflowTask
from conductor.client.workflow.task.task import TaskInterface
from conductor.client.workflow.task.task_type import TaskType


class AgentTask(TaskInterface):
    """Invoke a deployed agent using the server's native AGENT task."""

    def __init__(
        self,
        task_ref_name: str,
        agent_name: str,
        prompt: str,
    ) -> None:
        super().__init__(
            task_reference_name=task_ref_name,
            task_type=TaskType.USER_DEFINED,
            task_name="invoke_agent",
            input_parameters={
                "agentType": "conductor",
                "name": agent_name,
                "prompt": prompt,
                "pollIntervalSeconds": 5,
                "maxDurationSeconds": 300,
            },
        )

    def to_workflow_task(self) -> WorkflowTask:
        # The SDK requires a TaskType enum but does not yet include AGENT.
        # Replace the placeholder only in the definition sent to the server.
        task = super().to_workflow_task()
        task.type = "AGENT"
        return task

File: webhooks/java/src/main/java/io/orkes/codelab/webhooks/utils/AgentTask.java
package io.orkes.codelab.webhooks.utils;

import com.netflix.conductor.common.metadata.tasks.TaskType;
import com.netflix.conductor.common.metadata.workflow.WorkflowTask;
import com.netflix.conductor.sdk.workflow.def.tasks.Task;

/**
 * Workflow adapter for invoking a deployed Conductor agent with the server's native AGENT task, so
 * it can be added to a workflow alongside other tasks.
 */
public final class AgentTask extends Task<AgentTask> {

    public AgentTask(String taskReferenceName, String agentName, String prompt) {
        // The SDK's TaskType enum has no AGENT member, so SIMPLE is a placeholder that
        // updateWorkflowTask replaces with the type the server expects.
        super(taskReferenceName, TaskType.SIMPLE);
        name("invoke_agent");
        // Run the agent deployed to this Conductor cluster under agentName. The
        // default agentType, "a2a", calls an external A2A agent instead.
        input("agentType", "conductor");
        input("name", agentName);
        input("prompt", prompt);
        // How often, in seconds, the task checks on the agent, and how long the
        // agent may run before the task fails and Conductor cancels the agent.
        input("pollIntervalSeconds", 5);
        input("maxDurationSeconds", 300);
    }

    @Override
    protected void updateWorkflowTask(WorkflowTask workflowTask) {
        workflowTask.setType("AGENT");
    }
}
